src/index.ts

```diff
@@ -506,7 +506,11 @@ function validate(...args: string[]): void {
     if (arg.split('-').length - 1 > 1) {
       throw new Error(`Invalid number: more than one negative sign '${arg}'`)
     }
-    if (/[^\d.\-x]/.test(arg)) {
+    if (
+      /^(?!\s*-?(0x[0-9A-Fa-f]+|(\d+(\.\d*)?|\.\d+)([eE][+-]?\d+)?\s*)$).+/.test(
```

Comment:
Use `!/^-?(?:0x[0-9A-Fa-f]+|(?:\d+(?:\.\d*)?|\.\d+))$/.text(arg)` which removes scientific notation, doesn't use a negative look ahead, and removes the white space checks.